api/hypershift/v1beta1/azure.go

```diff
@@ -349,25 +354,25 @@ type AzurePlatformSpec struct {
 	//          ResourceGroupName is <resourceGroupName>.
 	//
 	// +kubebuilder:default:=default
-	// +kubebuilder:validation:Required
+	// +required
 	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9_()\-\.]{1,89}[a-zA-Z0-9_()\-]$`
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceGroupName is immutable"
+	// +kubebuilder:validation:MaxLength=90
 	// +immutable
-	// +required
 	ResourceGroupName string `json:"resourceGroup"`
 
-	// VnetID is the ID of an existing VNET to use in creating VMs. The VNET can exist in a different resource group
+	// vnetID is the ID of an existing VNET to use in creating VMs. The VNET can exist in a different resource group
 	// other than the one specified in ResourceGroupName, but it must exist under the same subscription as
 	// SubscriptionID.
 	//
 	// In ARO HCP, this will be the ID of the customer provided VNET.
 	//
 	// Example: /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>
 	//
-	// +kubebuilder:validation:Required
+	// +optional
```

Comment:
Why was this reverted to optional? It is required.

Reply:
I'll remove omitempty instead.